crates/router/src/core/payment_methods.rs

```diff
@@ -2578,7 +2584,18 @@ pub async fn payment_methods_session_confirm(
             tokenization_type: common_enums::TokenizationType::SingleUse,
             ..
         }) => {
-            todo!("single use tokenization are not implemented")
+            Box::pin(add_token_call_to_store(
```

Comment:
instead of branching can we this polymorphic by the `TokenizationType`?

And move the branching based things to a new function, avoid branching in handler/core functions.